src/test/java/core/basesyntax/FruitShopServiceTest.java

```diff
@@ -0,0 +1,82 @@
+package core.basesyntax;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Storage;
+import core.basesyntax.model.dto.FruitRecordDto;
+import core.basesyntax.service.AdditionStrategy;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.service.FruitShopServiceImpl;
+import core.basesyntax.service.Operation;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.ReduceStrategy;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitShopServiceTest {
+    private static List<FruitRecordDto> expectedTransactionDtos;
+    private static FruitShopService fruitService;
+    private static final AdditionStrategy additionStrategy = new AdditionStrategy();
+
+    @After
+    public void clear() {
+        Storage.fruits.clear();
+    }
+
+    @BeforeClass
+    public static void setUp() {
+        fillTransactionDto();
```

Comment:
```suggestion
        Map<Operation, OperationStrategy> operationStrategyMap = new HashMap<>();
        operationStrategyMap.put(Operation.BALANCE, additionStrategy);
        operationStrategyMap.put(Operation.SUPPLY, additionStrategy);
        operationStrategyMap.put(Operation.RETURN, additionStrategy);
        operationStrategyMap.put(Operation.PURCHASE, new ReduceStrategy());

        fillTransactionDto();
```
not all of the code, just split it to make more readable